runtime/vm/jnimisc.cpp

```diff
@@ -515,11 +515,12 @@ newStringUTF(JNIEnv *env, const char *bytes)
 			/* Slow case -- verify the encoding, fix if necessary.
 			 * String may double in size (invalid singles becoming doubles).
 			 */
-			data = (U_8*)jniArrayAllocateMemoryFromThread(currentThread, length * 2);
+			data = (U_8*)jniArrayAllocateMemoryFromThread(currentThread, length * 2 + 1);
```

Comment:
We don't need to include the null terminator as java strings are not null terminated and this is allocating a temporary array to hold the backing data